### ISSUE TITLE
Deleting an application in GKE Autopilot fails

### ISSUE THREAD

**Issue by robertlagrant** [argoproj/argo-cd#5813]
**Describe the bug**
Cannot delete an application created in GKE Autopilot using the Argo UI.

**To Reproduce**
Having created the guestbook-helm example app using the UI in Google Kubernetes Engine (Autopilot mode), then deleting the guestbook-helm example app results in the following error:
```
failed to load initial state of resource MutatingWebhookConfiguration.admissionregistration.k8s.io: mutatingwebhookconfigurations.admissionregistration.k8s.io is forbidden: User "system:serviceaccount:argocd:argocd-application-controller" cannot list resource "mutatingwebhookconfigurations" in API group "admissionregistration.k8s.io" at the cluster scope: GKEAutopilot authz: cluster scoped resource "mutatingwebhookconfigurations/" is managed and access is denied
```
**Expected behavior**
The application should delete

**Screenshots**
(The cluster was ephemeral, so the IP is fine to show here.)
![image](https://user-images.githubusercontent.com/1766161/111709261-2df0b900-883f-11eb-83b7-9eb14ebe8eae.png)

**Version**
I didn't use ArgoCD, just a kubectl install of ArgoCD and then used the UI, but still. Here you go!
```shell
% argocd version
argocd: v1.8.7+eb3d1fb.dirty
  BuildDate: 2021-03-07T19:33:57Z
  GitCommit: eb3d1fb84b9b77cdffd70b14c4f949f1c64a9416
  GitTreeState: dirty
  GoVersion: go1.16
  Compiler: gc
  Platform: darwin/amd64
FATA[0000] Argo CD server address unspecified
```

**Comment by yuya-takeyama** [comment]
I'm facing the same error.
Actually, the error occurs when I create an Application.

Seems it's related to this limitation of Autopilot:
https://cloud.google.com/kubernetes-engine/docs/concepts/autopilot-overview#webhooks_limitations
>Webhooks limitations
You cannot create custom mutating admission webhooks for Autopilot clusters, but you can create custom validating webhooks.

**Comment by jannfis** [comment]
I think you might want to exclude `MutatingWebhookConfiguration` resources  from Argo CD when running with GKE Autopilot, so that Argo CD will completely ignore them: https://argo-cd.readthedocs.io/en/stable/operator-manual/declarative-setup/#resource-exclusioninclusion

**Comment by KeisukeYamashita** [comment]
@jannfis  I configured this filter to a existing ArgoCD but still having issues with deleting an app...

**Comment by xpicio** [comment]
Hello, i have the same issue on an autopilot cluster with 1.19.9-gke.1400. I'm facing the issue when creating the app, not on deleting. The argocd version is:

```
argocd: v2.0.1+33eaf11.dirty
  BuildDate: 2021-04-21T18:44:28Z
  GitCommit: 33eaf11e3abd8c761c726e815cbb4b6af7dcb030
  GitTreeState: dirty
  GoVersion: go1.16.3
  Compiler: gc
  Platform: darwin/amd64
argocd-server: v2.0.2+9a7b0bc
  BuildDate: 2021-05-20T19:30:25Z
  GitCommit: 9a7b0bc35048835516fc5f5f31a93d4c1bd61242
  GitTreeState: clean
  GoVersion: go1.16
  Compiler: gc
  Platform: linux/amd64
  Ksonnet Version: v0.13.1
  Kustomize Version: v3.9.4 2021-02-09T19:22:10Z
  Helm Version: v3.5.1+g32c2223
  Kubectl Version: v0.20.4
  Jsonnet Version: v0.17.0
```

Adding the follow snippet on argocd-cm ConfigMap (as suggested by @jannfis)

```
  resource.exclusions: |
    - apiGroups:
      - "*"
      kinds:
      - "MutatingWebhookConfiguration"
      clusters:
      - "*"
```

I can create the app, but what argocd's features i will miss with that configuration ?

**Comment by xpicio** [comment]
any news about it ? i would just understand if it is a good, or not, idea use argocd on autopilot cluster.

**Comment by tschmidt64-SC** [comment]
+1 have this same issue

**Comment by grahammather** [comment]
This appears to be fixed in Kubernetes version 1.21, which google autopilot will install if you specify [Rapid]( https://cloud.google.com/kubernetes-engine/docs/release-notes) as your release channel. I tried to `argocd-autopilot repo bootstrap` on a 1.20 cluster and got the error listed in this issue. Recreated the cluster with the rapid release channel, tried again, and the error did not show up. The [Autopilot overview](https://cloud.google.com/kubernetes-engine/docs/concepts/autopilot-overview#webhooks_limitations) indicates that 1.21 introduced the ability to use mutating admission webhooks.

**Comment by jgwest** [comment]
Based on comments here, it's resolved by a newer version of GKE, and the versions of Kubernetes that are mentioned are out of support. Is anyone still able to reproduce this?
